### PR TITLE
Fixed number truncation in displayed values

### PR DESCRIPTION
**Pull request by jjjoness**
Can now handle truncation of all numbers and all decimal places correctly.

Signed-off-by: John Jones-Steele <jjjoness@amazon.com>

### CODE REVIEW

File: Code/Framework/AzQtComponents/AzQtComponents/Utilities/Conversions.cpp [hunk 1]
```diff
@@ -43,19 +43,20 @@ namespace AzQtComponents
...
-        // so we can remove the last value
-        QString retValue = locale.toString(value, 'f', (numDecimals > 0) ? numDecimals + 1 : 0);
+        // We want to truncate, not round. toString will round, so we add extra decimal places to the formatting
+        // so we can remove the last values
+        QString retValue = locale.toString(value, 'f', (numDecimals > 0) ? numToStringDecimals : 0);
 
         // Handle special cases when we have decimals in our value
         if (numDecimals > 0)
         {
             // Truncate the extra digit now, if it's still there
```
Comment (mbalfour-amzn) [inline Code/Framework/AzQtComponents/AzQtComponents/Utilities/Conversions.cpp:55]: Nit: This should be plural now with the change, right?  "Truncate the extra digits now, if they're still there"

**Review by mbalfour-amzn — APPROVED** [review]

File: Code/Framework/AzQtComponents/AzQtComponents/Utilities/Conversions.cpp (hunk 1 above)
Comment (jjjoness) [reply Code/Framework/AzQtComponents/AzQtComponents/Utilities/Conversions.cpp]: Ooops. Change made.

File: Code/Framework/AzQtComponents/AzQtComponents/Utilities/Conversions.cpp [hunk 2]
```diff
@@ -43,19 +43,20 @@ namespace AzQtComponents
     {
         const QChar decimalPoint = locale.decimalPoint();
         const QChar zeroDigit = locale.zeroDigit();
+        const int numToStringDecimals = numDecimals < 8 ? 8 : numDecimals;
```
Comment (hultonha) [inline Code/Framework/AzQtComponents/AzQtComponents/Utilities/Conversions.cpp:46]: nit/suggestion: Could this be written as `const int numToStringDecimals = AZStd::max(numDecimals, 8);`?

**Review by hultonha — COMMENTED** [review]
Would it be possible to please add some tests to this with the inputs you tried to make sure we get the correct formatted output, this seems like a great candidate for a unit test or two 🙂

File: Code/Framework/AzQtComponents/AzQtComponents/Utilities/Conversions.cpp (hunk 2 above)
Comment (jjjoness) [reply Code/Framework/AzQtComponents/AzQtComponents/Utilities/Conversions.cpp:46]: WIll make the change and add a new UnitTest.

**Review by hultonha — APPROVED** [review]
Awesome! Thanks John, looking good :+1:

**Pull request merged**